packages/react/src/components/cookieConsent/modal/Modal.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useContext, useEffect, useState } from 'react';
+import ReactDOM from 'react-dom';
+import { VisuallyHidden } from '@react-aria/visually-hidden';
+
+import classNames from '../../../utils/classNames';
+import styles from '../CookieConsent.module.scss';
+import { CookieConsentContext, useCookieConsentUiTexts, useFocusShift } from '../CookieConsentContext';
+import { Content } from '../content/Content';
+import { useEscKey } from '../useEscKey';
+
+export function Modal(): React.ReactElement | null {
+  const cookieConsentContext = useContext(CookieConsentContext);
+  const hasOnlyRequiredConsents =
+    !cookieConsentContext.content.optionalCookies || cookieConsentContext.content.optionalCookies.groups.length === 0;
+  const shouldShowModal = !hasOnlyRequiredConsents && !cookieConsentContext.hasUserHandledAllConsents();
+  const [isModalInitiallyShown] = useState<boolean>(shouldShowModal);
+  const [popupTimerComplete, setPopupTimerComplete] = useState<boolean>(false);
+  const popupDelayInMs = 500;
+  // if hasUserHandledAllConsents() was false at first and then later true, user must have saved consents.
+  const showScreenReaderSaveNotification = isModalInitiallyShown && !shouldShowModal;
+  const { settingsSaved } = useCookieConsentUiTexts();
+  const containerId = 'HdsCookieConsentContainer';
+  const getContainerElement = (): HTMLElement | null => document.getElementById(containerId);
+  const [isDomReady, setIsDomReady] = useState<boolean>(false);
+
+  useEffect(() => {
+    setTimeout(() => setPopupTimerComplete(true), popupDelayInMs);
+  }, []);
+
+  useEffect(() => {
+    const containerElement = getContainerElement();
+    if (shouldShowModal && !isDomReady) {
```

Comment:
Maybe some combined state would be better than resolve these from two different states

Reply:
Outdated. Modal has no domready anymore